File: core/src/com/mygdx/game/components/TransformComponent.java
package com.mygdx.game.components;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.math.Vector3;

public class TransformComponent implements Component {
    public final Vector3 pos = new Vector3();
    public Vector2 scale = new Vector2(0.5f, 0.5f);
    public float rotation = 0.0f;
}

File: core/src/adictive/games/components/TransformComponent.kt
package adictive.games.components

import com.badlogic.ashley.core.Component
import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.math.Vector3

class TransformComponent : Component {
    val pos = Vector3()
    val size = Vector2()
    val lastPos = Vector3(0f, 0f, 0f)
    val scale = Vector2(1f, 1f)
    var rotation = 0.0f
}
